api/config/config.go

```diff
@@ -23,9 +39,38 @@ func init() {
 // DefaultConfig defines the default configuration options for api
 func DefaultConfig() Config {
 	return Config{
-		StartGrpcServer: defaultStartGRPCServer, // note: all bool flags default to false so don't set one of these to true here
-		GrpcServerPort:  defaultGRPCServerPort,
-		StartJSONServer: defaultStartJSONServer,
-		JSONServerPort:  defaultJSONServerPort,
+		StartGrpcServer:    defaultStartGRPCServer, // note: all bool flags default to false so don't set one of these to true here
+		StartGrpcServices:  nil,                    // note: cannot configure an array as a const
+		GrpcServerPort:     defaultGRPCServerPort,
+		NewGrpcServerPort:  defaultNewGRPCServerPort,
+		StartJSONServer:    defaultStartJSONServer,
+		StartNewJSONServer: defaultStartNewJSONServer,
+		JSONServerPort:     defaultJSONServerPort,
+		NewJSONServerPort:  defaultNewJSONServerPort,
+		StartNodeService:   defaultStartNodeService,
+		StartMeshService:   defaultStartMeshService,
+	}
+}
+
+// ParseServicesList enables the requested services
+func (s *Config) ParseServicesList() error {
```

Comment:
general question: why don't we simply start all services? Does it cost any resources to have these endpoints available at all times?

Reply:
It's not about resources, it's more of a security thing. A node operator may only want to expose a certain set of endpoints, and not others - e.g., if you're running a node that's serving third parties, you would probably *not* want to expose the NodeService or the SmesherService, since that would allow third parties to control your node, shut it down, etc. Does that make sense?